clippy_lints/src/cargo/multiple_crate_versions.rs

```diff
@@ -16,7 +16,7 @@ pub(super) fn check(cx: &LateContext<'_>, metadata: &Metadata) {
 
     if let Some(resolve) = &metadata.resolve
         && let Some(local_id) = packages.iter().find_map(|p| {
-            if p.name == local_name.as_str() {
+            if p.name.replace('-', "_") == local_name.as_str() {
```

Comment:
I wonder if we can avoid the allocation by using iteration here

I wonder if that's necessary though

Reply:
Reading more: crates.io only allows a subset of ASCII:
https://github.com/rust-lang/crates.io/blob/825cce6c189fcf1174da4b7d31855bae430ed97c/src/models/keyword.rs#L56-L63

Cargo itself only allows ASCII:
https://github.com/rust-lang/cargo/blob/92395d90106b3b61bcb68bcf2069052c93771764/src/cargo/ops/cargo_new.rs#L260C32-L266

But I still believe `char` is better.

Reply:
> While this allocates a char on each end per char compared at least it stops once they're not equal.

it's a stack allocation, not usually what people mean when talking about allocations.

Reply:
but yeah why not make the change to use `.as_bytes()` and normalize `b'-'` to `b'_'`

Reply:
> but yeah why not make the change to use `.as_bytes()` and normalize `b'-'` to `b'_'`

Let me do that.

Reply:
@Manishearth Done. Thanks for the mentoring.